python/cuml/svm/svm.pyx

```diff
@@ -0,0 +1,297 @@
+            self.kernel_params = <size_t> kernel_params
+        else:
+            kernel_params = <KernelParams*><size_t> self.kernel_params
+
+        if self.dtype == np.float32:
+            if self.svcHandle is None:
+                svc_f = new CppSVC[float](
+                    handle_[0], self.C, self.tol, deref(kernel_params),
+                    self.cache_size, self.max_iter)
+                self.svcHandle = <size_t> svc_f
+            else:
+                svc_f = <CppSVC[float]*><size_t> self.svcHandle
+            svc_f.fit(<float*>X_ptr, <int>self.n_rows,
+                      <int>self.n_cols, <float*>y_ptr)
+            self.intercept_ = svc_f.b
+            self.n_support_ = svc_f.n_support
+
+        else:
+            if self.svcHandle is None:
+                svc_d = new CppSVC[double](
+                    handle_[0], self.C, self.tol, deref(kernel_params),
+                    self.cache_size, self.max_iter)
+                self.svcHandle = <size_t> svc_d
+            else:
+                svc_d = <CppSVC[double]*><size_t> self.svcHandle
+            svc_d.fit(<double*>X_ptr, <int>self.n_rows, <int>self.n_cols,
+                      <double*>y_ptr)
+            self.intercept_ = svc_d.b
+            self.n_support_ = svc_d.n_support
+
+        self.handle.sync()
+
+        del X_m
+        del y_m
+
+        return self
+
+    def predict(self, X):
+        """
+        Predicts the y for X.
+
+        Parameters
+        ----------
+        X : array-like (device or host) shape = (n_samples, n_features)
+            Dense matrix (floats or doubles) of shape (n_samples, n_features).
+            Acceptable formats: cuDF DataFrame, NumPy ndarray, Numba device
+            ndarray, cuda array interface compliant array like CuPy
+        Returns
+        ----------
+        y: cuDF Series
+           Dense vector (floats or doubles) of shape (n_samples, 1)
+        """
+
+        if self.svcHandle is None:
+            raise NotFittedError("Call fit before prediction")
+
+        cdef uintptr_t X_ptr
+        X_m, X_ptr, n_rows, n_cols, pred_dtype = \
+            input_to_dev_array(X, check_dtype=self.dtype)
+
+        preds = cudf.Series(zeros(n_rows, dtype=self.dtype))
+        cdef uintptr_t preds_ptr = get_cudf_column_ptr(preds)
+
+        cdef CppSVC[float]* svc_f
+        cdef CppSVC[double]* svc_d
+
+        if self.dtype == np.float32:
+            svc_f = <CppSVC[float]*><size_t> self.svcHandle
+            svc_f.predict(<float*>X_ptr,
+                          <int>n_rows,
+                          <int>n_cols,
+                          <float*>preds_ptr)
+        else:
+            svc_d = <CppSVC[double]*><size_t> self.svcHandle
+            svc_d.predict(<double*>X_ptr,
+                          <int>n_rows,
+                          <int>n_cols,
+                          <double*>preds_ptr)
+
+        self.handle.sync()
+
+        del(X_m)
+
```

Comment:
I *believe* you may need to delete `preds` here as well (to reduce the reference count).